File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cs
public class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        int first = arr[0];
        answer = LCM(first , arr[1]);
        
        for(int i = 1; i < arr.Length - 1; i++)
        {
            answer = LCM(answer, arr[i + 1]);
        }
        return answer;
    }
    
    private int GCD(int a, int b)
    {
        while(b != 0)
        {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    private int LCM(int a, int b)
    {
        return (a * b) / GCD(a, b);
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(int a, int b)
{
    if(a == 0) return b;
    return gcd(b % a, a); 
}

int lcm(int a, int b)
{
    return (a * b) / gcd(a, b);
}

int solution(vector<int> arr) {
    int answer = arr[0];
    
    for(int i = 1; i < arr.size(); i++)
    {
        answer = lcm(answer, arr[i]);
    }    
    
    return answer;
}